api/tools/src/main/java/org/pmiops/workbench/tools/OrphanedProjects.java

```diff
@@ -0,0 +1,560 @@
+
+  private static final Option ORGANIZATION_OPT =
+      Option.builder()
+          .longOpt("organization")
+          .desc(
+              "The Google Organization to query for projects.  firecloud.org and test.firecloud.org are supported.")
+          .required()
+          .hasArg()
+          .build();
+
+  private static final Option RW_ENV_OPT =
+      Option.builder()
+          .longOpt("environment")
+          .desc("The AoU environment to access for project cleanup.")
+          .required()
+          .hasArg()
+          .build();
+
+  private static final Option TERRA_ADMIN_TOKEN_OPT =
+      Option.builder()
+          .longOpt("terra-admin-token")
+          .desc("A current bearer token of a Terra admin, used to query for google projects.")
+          .required()
+          .hasArg()
+          .build();
+
+  private static final Options OPTIONS =
+      new Options()
+          .addOption(ORGANIZATION_OPT)
+          .addOption(RW_ENV_OPT)
+          .addOption(TERRA_ADMIN_TOKEN_OPT);
+
+  private void processTier(
+      CloudResourceManager manager,
+      List<Folder> folders,
+      String environment,
+      String tierName,
+      Collection<FieldsForOrphanChecking> workspaces) {
+    LOG.info(
+        String.format(
+            "Retrieved %d active workspaces in the %s tier from the AoU DB",
+            workspaces.size(), tierName));
+
+    String folderName = getFolderNameForTier(environment, tierName);
+    Folder folder =
+        folders.stream()
+            .filter(f -> f.getDisplayName().equals(folderName))
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new BadRequestException(
+                        String.format(
+                            "Folder name %s not found for tier %s", folderName, tierName)));
+
+    List<Project> projects = OPUtils.getProjectsInFolder(manager, folder);
+
+    List<Pair<FieldsForOrphanChecking, Project>> matches = findMatches(workspaces, projects);
+
+    Set<FieldsForOrphanChecking> dbMatched =
+        matches.stream().map(Pair::getLeft).collect(Collectors.toSet());
+    Set<Project> cloudMatched = matches.stream().map(Pair::getRight).collect(Collectors.toSet());
+
+    Set<FieldsForOrphanChecking> dbOnly = Sets.difference(new HashSet<>(workspaces), dbMatched);
+    Set<Project> cloudOnly = Sets.difference(new HashSet<>(projects), cloudMatched);
+
+    LOG.info(
+        String.format(
+            "%d projects match, %d projects are only in the DB, and %d projects are only in the cloud",
+            matches.size(), dbOnly.size(), cloudOnly.size()));
+
+    Path outputDir = makeOutputDir();
+    OPUtils.writeMatchesToTierCSV(outputDir, tierName, matches);
+    OPUtils.writeWorkspacesToTierCSV(outputDir, tierName, dbOnly);
+    OPUtils.writeProjectsToTierCSV(outputDir, tierName, cloudOnly);
+
+    // can't use Multimap.index() because some projects are null
+    Map<Optional<String>, List<String>> dbOnlyProjectsByPrefix =
```

Comment:
splitting by project name prefix because I expect that we will find different patterns for each.  This also splits Test from Local, which we'll need for future analysis.